src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] hashTable;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        hashTable = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int)(DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putIfKeyIsNull(value);
+            return;
+        }
+        int hash = hashCode(key);
+        int index = indexFor(hash, hashTable.length);
 
+        for (Node<K, V> node = hashTable[index]; node != null; node = node.next) {
+            if (node.hash == hash && (key.equals(node.key))) {
+                node.value = value;
+                return;
+            }
+        }
```

Comment:
you have almost the same duplicated code, you can create new private method for it to avoid duplication 
the method should return node by key that you might reuse it.